src/main/java/com/exasol/spark/common/SchemaConverter.java

```diff
@@ -0,0 +1,115 @@
+        return new StructType(fields.toArray(new StructField[0]));
+    }
+
+    /**
+     * Converts a {@link ColumnDescription} column description into a Spark {@link StructField} field type.
+     *
+     * @param column column description
+     * @return matching Spark StructField type
+     */
+    public StructField convertColumn(final ColumnDescription column) {
+        return DataTypes.createStructField(column.getName(), mapJDBCType(column), column.isNullable());
+    }
+
+    private DataType mapJDBCType(final ColumnDescription column) {
+        final int jdbcType = column.getType();
+        switch (jdbcType) {
+        case Types.BIT:
+        case Types.BOOLEAN:
+            return DataTypes.BooleanType;
+        case Types.TINYINT:
+        case Types.SMALLINT:
+            return DataTypes.ShortType;
+        case Types.DOUBLE:
+        case Types.FLOAT:
+            return DataTypes.DoubleType;
+        case Types.REAL:
+            return DataTypes.FloatType;
+        case Types.INTEGER:
+            return column.isSigned() ? DataTypes.IntegerType : DataTypes.LongType;
+        case Types.BIGINT:
+            return column.isSigned() ? DataTypes.LongType : DataTypes.createDecimalType(20, 0);
+        case Types.DECIMAL:
+        case Types.NUMERIC:
+            if (column.getPrecision() != 0 || column.getScale() != 0) {
+                return createDecimalDataType(column.getPrecision(), column.getScale());
+            } else {
+                return DecimalType.USER_DEFAULT();
+            }
+        case Types.CHAR:
+        case Types.NCHAR:
+        case Types.VARCHAR:
+        case Types.NVARCHAR:
+        case Types.LONGVARCHAR:
+        case Types.LONGNVARCHAR:
+        case Types.BINARY:
+        case Types.VARBINARY:
+        case Types.LONGVARBINARY:
+            return DataTypes.StringType;
+        case Types.DATE:
+            return DataTypes.DateType;
+        case Types.TIME:
+        case Types.TIMESTAMP:
+            return DataTypes.TimestampType;
+        default:
+            throw new UnsupportedOperationException(
+                    ExaError.messageBuilder("E-SCCJ-4").message("Unsupported JDBC type {{type}} for column {{column}}.",
+                            column.getType(), column.getName()).ticketMitigation().toString());
+        }
+    }
+
+    private DataType createDecimalDataType(int precision, int scale) {
+        if (precision > DecimalType.MAX_PRECISION()) {
+            throw new IllegalStateException(ExaError.messageBuilder("E-SCCJ-5") //
+                    .message("Exasol decimal precision {{exasolPrecision}} is larger than "
+                            + "maximum Spark decimal precision {{sparkPrecision}}.") //
+                    .parameter("exasolPrecision", precision) //
+                    .parameter("sparkPrecision", DecimalType.MAX_PRECISION()) //
+                    .ticketMitigation() //
```

Comment:
```suggestion
```
We can't do anything there. Please add a note to the error message that this is not supported.